### PR TITLE
Add Scope Validation Integration testcases

### PR DESCRIPTION
**Pull request by sahandilshan**
## Purpose

* Add scope validation integration test cases.

## Related PR(s)

https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1766
**(Note: Please merge above PR before merging this PR)**

### CODE REVIEW

**Comment by janakamarasena** [comment]
Since in the fix we removed the consent prompting for`SYSTEM` scope lets add a assertion to make sure that it is not present.

**Comment by sahandilshan** [comment]
> Since in the fix we removed the consent prompting for`SYSTEM` scope lets add a assertion to make sure that it is not present.

Hi @janakamarasena, This has been fixed with https://github.com/wso2/product-is/pull/13004/commits/3f710a27f527bb6b087d1ced95d92637bc6233eb

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCScopeValidationTest.java [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+        initUser(scope);
+        createUser(user);
+        userInfo.setUserName(user.getUsername());
+        userInfo.setPassword(user.getPassword());
+
+        playgroundApp = initApplication();
+        serviceProvider = createApplication(new ServiceProvider(), playgroundApp);
+        client = HttpClientBuilder.create().setDefaultCookieStore(cookieStore).build();
+
+
```
Comment (chamathns) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCScopeValidationTest.java:74]: redundant blank lines.
Let's check other places as well

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCScopeValidationTest.java [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+        List<NameValuePair> urlParameters = new ArrayList<NameValuePair>();
+        urlParameters.add(new BasicNameValuePair("grantType", OAuth2Constant.OAUTH2_GRANT_TYPE_CODE));
+        urlParameters.add(new BasicNameValuePair("consumerKey", application.getClientId()));
+        urlParameters.add(new BasicNameValuePair("callbackurl", application.getCallBackURL()));
+        urlParameters.add(new BasicNameValuePair("authorizeEndpoint", OAuth2Constant.APPROVAL_URL));
+        urlParameters.add(new BasicNameValuePair("authorize", OAuth2Constant.AUTHORIZE_PARAM));
+        urlParameters.add(new BasicNameValuePair("scope", OAuth2Constant.OAUTH2_SCOPE_OPENID + " " +
+                OAuth2Constant.OAUTH2_SCOPE_EMAIL + " " + OAuth2Constant.OAUTH2_SCOPE_PROFILE + " " +
+                INTERNAL_LOGIN_SCOPE + " " + INTERNAL_USER_MGT_UPDATE_SCOPE + " " + INVALID_SCOPE + " " +
+                SYSTEM_SCOPE));
```
Comment (chamathns) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCScopeValidationTest.java:247]: Isn't there's a util method that does this? Let's try to reuse it and modify the "scope" param

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCScopeValidationTest.java [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+
+        playgroundApp = new OIDCApplication(OIDCUtilTest.playgroundAppOneAppName,
+                OIDCUtilTest.playgroundAppOneAppContext,
+                OIDCUtilTest.playgroundAppOneAppCallBackUri);
+        playgroundApp.addRequiredClaim(OIDCUtilTest.emailClaimUri);
+        playgroundApp.addRequiredClaim(OIDCUtilTest.firstNameClaimUri);
+        return playgroundApp;
+    }
+
+    private void testSendAuthenticationRequest(OIDCApplication application, HttpClient client)
```
Comment (chamathns) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCScopeValidationTest.java:144]: Let's use a data provider for this and include different sets of scopes

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCScopeValidationTest.java [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+                "'" + INTERNAL_USER_MGT_UPDATE_SCOPE + "' scopes should not exist after the validation. " +
+                        "Since user does not have permission for this scope.");
+        Assert.assertFalse(scopes.contains(INVALID_SCOPE), "'" + INVALID_SCOPE + "' scopes should not exist" +
+                " after the validation. This scope does not exist.");
+
+    }
+
+    @Test(groups = "wso2.is", dependsOnMethods = {"testScopeValidationWithNormalUser"}, description = "Test scope" +
+            " validation with an admin user.")
+    public void testScopeValidationWithAdminUser() throws Exception {
```
Comment (chamathns) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCScopeValidationTest.java:107]: Can't we use a data provider for this and include different sets of scopes?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCScopeValidationTest.java [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+        user.addUserClaim(OIDCUtilTest.firstNameClaimUri, OIDCUtilTest.firstName);
+        user.addUserClaim(OIDCUtilTest.lastNameClaimUri, OIDCUtilTest.lastName);
+        user.addRole(role);
+    }
+
+    private OIDCApplication initApplication() {
+
+        playgroundApp = new OIDCApplication(OIDCUtilTest.playgroundAppOneAppName,
+                OIDCUtilTest.playgroundAppOneAppContext,
+                OIDCUtilTest.playgroundAppOneAppCallBackUri);
```
Comment (chamathns) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCScopeValidationTest.java:138]: The formatting here looks off. Please check

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCScopeValidationTest.java (hunk 3 above)
Comment (chamathns) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCScopeValidationTest.java:144]: Let's rename the private methods without the 'test' prefix. Can be confused with the actual test methods

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCScopeValidationTest.java (hunk 2 above)
Comment (sahandilshan) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCScopeValidationTest.java:247]: There's util method available to do this. But that method only adds `openid`, `email`, and `profile` scopes. Here I need to add some additional scopes as well, That's why I extracted that code snippet from the util class. I cannot add additional scopes after the object has been added to the list.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCScopeValidationTest.java (hunk 3 above)
Comment (sahandilshan) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCScopeValidationTest.java]: Fixed with https://github.com/wso2/product-is/pull/13004/commits/6215fdfab95c193b90112d9a983ea3bd0113ee46

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCScopeValidationTest.java (hunk 5 above)
Comment (sahandilshan) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCScopeValidationTest.java]: Fixed with https://github.com/wso2/product-is/pull/13004/commits/6215fdfab95c193b90112d9a983ea3bd0113ee46

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCScopeValidationTest.java (hunk 1 above)
Comment (sahandilshan) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCScopeValidationTest.java]: Fixed with https://github.com/wso2/product-is/pull/13004/commits/6215fdfab95c193b90112d9a983ea3bd0113ee46

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCScopeValidationTest.java (hunk 4 above)
Comment (sahandilshan) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCScopeValidationTest.java]: Fixed with https://github.com/wso2/product-is/pull/13004/commits/a0ee361a76ed7d6c50f7c58215fff138992e1fbc

**Review by chamathns — APPROVED** [review]

**Comment by sahandilshan** [comment]
Closing this PR since we need to update this PR with the changes to this issue https://github.com/wso2/product-is/issues/13024

**Pull request closed without merging**